src/js/helpers/Util.js

```diff
@@ -113,6 +113,26 @@ var Util = {
 
     return obj;
   },
+  objectCreateWithAdder: function (valuePrefix = "") {
+    const obj = {};
+
+    // Make the properties of the object read-only, but the object extensible.
+    Object.defineProperty(obj, "add", {
+      enumerable: false,
+      configurable: false,
+      writable: false,
+      value: function (key) {
+        Object.defineProperty(obj, key, {
+          enumerable: false,
+          configurable: false,
+          writable: false,
+          value: valuePrefix + key
```

Comment:
Please refactor this method as proposed in https://github.com/mesosphere/marathon-ui/pull/596/files#r52300585